s2spy/_base_calendar.py

```diff
@@ -52,34 +54,52 @@ def _get_anchor(self, year: int) -> pd.Timestamp:
         Returns:
             pd.Timestamp: Timestamp at the end of the anchor_years interval 0.
         """
-        return pd.Timestamp()
-
-    def set_max_lag(self, max_lag: int, allow_overlap: bool = False) -> None:
-        """Set the maximum lag of a calendar.
+        return pd.to_datetime(
+            f"{year}-" + self._anchor, format="%Y-" + self._anchor_fmt
+        )
 
-        Sets the maximum number of lag periods after the target period. If `0`,
-        the maximum lag will be determined by how many fit in each anchor year.
-        If a maximum lag is provided, the intervals can either only cover part
-        of the year, or extend over multiple years. In case of a large max_lag
-        number where the intervals extend over multiple years, anchor years will
-        be skipped to avoid overlapping intervals. To allow overlapping
-        intervals, use the `allow_overlap` kwarg.
+    def _parse_anchor(self, anchor_str: str) -> Tuple[str, str]:
+        """Parses the user-input anchor.
 
         Args:
-            max_lag: Maximum number of lag periods after the target period.
-            allow_overlap: Allows intervals to overlap between anchor years, if the
-                max_lag is set to a high enough number that intervals extend over
-                multiple years. `False` by default, to avoid train/test information
-                leakage.
+            anchor_str: Anchor string in the right formatting.
+
+        Returns:
+            Datetime formatter to parse the anchor into a date.
         """
-        if (max_lag < 0) or (max_lag % 1 > 0):
+        if re.fullmatch("\\d{1,2}-\\d{1,2}", anchor_str):
```

Comment:
From testing it seems we're missing a check for non-string data. Try giving `anchor=10` to a calendar. It will return the following error:
```py
TypeError: expected string or bytes-like object
```
Which is not very clear to users.

Reply:
I agree. I will add a check.